index.js

```diff
@@ -154,59 +156,137 @@ module.exports = {
+            (rootMsg) => {
+              foundMemberlessGroup = true
+              ssb.metafeeds.advanced.findById(
+                rootMsg.value.author,
+                (err, groupFeed) => {
+                  // prettier-ignore
+                  if (err) return cb(clarify(err, "failed finding details of the memberless group feed"))
 
-          ssb.db.create(
-            {
-              keys: groupFeed.keys,
-              content,
-              recps,
-              encryptionFormat: 'box2',
+                  rootMsg.value.content = rootMsg.meta.originalContent
+                  return cb(null, {
+                    groupInitMsg: rootMsg,
+                    groupFeed,
+                    myRoot,
+                  })
+                }
+              )
             },
-            (err, groupInitMsg) => {
+            (err) => {
               // prettier-ignore
-              if (err) return cb(clarify(err, 'Failed to create group init message when creating a group'))
-
-              const data = {
-                id: buildGroupId({
-                  groupInitMsg,
-                  groupKey: secret.toBuffer(),
-                }),
-                secret: secret.toBuffer(),
-                root: fromMessageSigil(groupInitMsg.key),
-                subfeed: groupFeed.keys,
+              if (err) return cb(clarify(err, "errored trying to find potential memberless feed"))
+              else if (!foundMemberlessGroup) {
+                return createGroupWithoutMembers(myRoot, cb)
               }
-
-              ssb.box2.addGroupInfo(data.id, {
-                key: data.secret,
-                root: data.root,
-              })
-
-              // Adding myself for recovery reasons
-              addMembers(data.id, [root.id], {}, (err) => {
-                // prettier-ignore
-                if (err) return cb(clarify(err, 'Failed to add myself to the group when creating a group'))
-
-                return cb(null, data)
-              })
             }
           )
-        })
+        )
       })
     }
 
+    function create(opts = {}, cb) {
+      if (cb === undefined) return promisify(create)(opts)
+
+      findOrCreateGroupWithoutMembers(
+        (err, { groupInitMsg, groupFeed, myRoot }) => {
+          // prettier-ignore
+          if (err) return cb(clarify(err, 'Failed to create group init message when creating a group'))
+
+          const secret = secretKeyFromPurpose(groupFeed.purpose)
+
+          const data = {
+            id: buildGroupId({
+              groupInitMsg,
+              groupKey: secret.toBuffer(),
+            }),
+            secret: secret.toBuffer(),
+            root: fromMessageSigil(groupInitMsg.key),
+            subfeed: groupFeed.keys,
+          }
+
+          ssb.box2.addGroupInfo(data.id, {
+            key: data.secret,
+            root: data.root,
+          })
+
+          // Adding myself for recovery reasons
+          addMembers(data.id, [myRoot.id], {}, (err) => {
```

Comment:
do we need to reindexencrypted after adding ourselves? didn't seem to help but here's a diff

```diff
diff --git a/index.js b/index.js
index 385bb86..fff02cb 100644
--- a/index.js
+++ b/index.js
@@ -281,7 +282,11 @@ module.exports = {
             // prettier-ignore
             if (err) return cb(clarify(err, 'Failed to add myself to the group when creating a group'))
 
-            return cb(null, data)
+            ssb.db.reindexEncrypted((err) => {
+              // prettier-ignore
+              if (err) return cb(clarify(err, "failed to reindexencrypted after adding self to newly created group"))
+              return cb(null, data)
+            })
           })
         }
       )

```